framework/src/fwk_arch.c

```diff
@@ -87,6 +89,18 @@ int fwk_arch_init(const struct fwk_arch_init_driver *driver)
         return FWK_E_PANIC;
     }
 
+    /*
+     * In case firmware running under other OS context, finish processing of
+     * any raised events/interrupts and return. Else continue to process events
+     * in a forever loop.
+     */
+#if defined(BUILD_HAS_SUB_SYSTEM_MODE)
+    fwk_process_event_queue();
+    (void)fwk_log_unbuffer();
```

Comment:
It would be better to wrap this all in a single function like it is being done with `__fwk_run_main_loop()` to improve readability

Reply:
With BUILD_HAS_SUB_SYSTEM_MODE, fwk_process_event_queue(), fwk_log_unbuffer() and fwk_arch_suspend() can be used in different place. That's why they are kept separated

Reply:
I know but it could be better if you wrap this two in a function like

```suggestion
void __fwk_run_main_single(void)
{
    fwk_process_event_queue();
    (void)fwk_log_unbuffer();
}
```